### PR TITLE
make the DHT announce flags a strong type

### PR DESCRIPTION
**Pull request by arvidn**

### CODE REVIEW

File: src/kademlia/dht_tracker.cpp [hunk 1]
```diff
@@ -321,10 +321,11 @@ namespace libtorrent { namespace dht {
 		, std::function<void(std::vector<tcp::endpoint> const&)> f)
 	{
 		for (auto& n : m_nodes)
-			n.second.dht.get_peers(ih, f, {}, false);
+			n.second.dht.get_peers(ih, f, {}, {});
```
Comment (aldenml) [inline src/kademlia/dht_tracker.cpp]: do you think it makes sense to add `::none` to `flags::bitfield_flag`at some point?

Comment (arvidn) [reply src/kademlia/dht_tracker.cpp]: I kind of like `{}`

Comment (arvidn) [reply src/kademlia/dht_tracker.cpp]: ( https://www.youtube.com/watch?v=NGrnKr9rSz4 )

Comment (aldenml) [reply src/kademlia/dht_tracker.cpp]: anja, :), looking at it

**Pull request merged**